File: src/main/java/Piruru/abstracts/PiruruStance.java
package Piruru.abstracts;

import com.megacrit.cardcrawl.core.CardCrawlGame;
import com.megacrit.cardcrawl.localization.StanceStrings;
import com.megacrit.cardcrawl.stances.AbstractStance;

import static Piruru.Piruru.makeID;

public abstract class PiruruStance extends AbstractStance {
    public String STANCE_ID;
    public StanceStrings stanceStrings;

    public PiruruStance() {
        super();
        STANCE_ID = makeID(this.getClass());
        ID = STANCE_ID;
        stanceStrings = CardCrawlGame.languagePack.getStanceString(ID);
        name = stanceStrings.NAME;
        updateDescription();
    }

}

File: src/main/kotlin/Piruru/abstracts/PiruruStance.kt
package Piruru.abstracts

import Piruru.Piruluk.Statics.makeID
import com.megacrit.cardcrawl.core.CardCrawlGame
import com.megacrit.cardcrawl.localization.StanceStrings
import com.megacrit.cardcrawl.stances.AbstractStance

abstract class PiruruStance : AbstractStance() {
    var STANCE_ID: String = makeID(this.javaClass)
    var stanceStrings: StanceStrings

    init {
        ID = STANCE_ID
        stanceStrings = CardCrawlGame.languagePack.getStanceString(ID)
        name = stanceStrings.NAME
        updateDescription()
    }

}
